caper/dict_tool.py

```diff
@@ -49,7 +53,16 @@ def flatten_dict(d, parent_key=()):
             items.extend(flatten_dict(v, parent_key=new_key).items())
         else:
             items.append((new_key, v))
-    return type(d)(items)
+    if reducer:
+        return {reducer.join(k): v for k, v in type(d)(items).items()}
+    else:
+        return type(d)(items)
+
+
+def flatten_dict_with_dot(d):
+    """Flattens dict with dot notation.
+    """
+    return {'.'.join(k): v for k, v in flatten_dict(d).items()}
```

Comment:
This `flatten_dict_with_dot` function seems unused/unnecessary? If so should be deleted.

Reply:
Good point. Deleted.